tests/Unit/Domain/CoordinateMaps/TestMapHelpers.hpp

```diff
@@ -338,6 +340,77 @@ void test_suite_for_map_on_unit_cube(const Map& map) {
   test_helper(map2);
 }
 
+/*!
+ * \ingroup TestingFrameworkGroup
+ * \brief Given a Map `map`, tests the map functions, including map inverse,
+ * jacobian, and inverse jacobian, for a series of points.
+ * These points are chosen in a sphere of radius `radius_of_sphere`, and the
+ * map is expected to be valid on the boundary of that sphere as well as
+ * in its interior.  The flag `include_origin` indicates whether to test the
+ * map at the origin.
+ * This test works only in 3 dimensions.
+ */
+template <typename Map>
+void test_suite_for_map_on_sphere(const Map& map,
+                                  const bool include_origin = true,
+                                  const double radius_of_sphere = 1.0) {
+  static_assert(Map::dim == 3, "Works only for a 3d map");
+
+  // Set up random number generator
+  const auto seed = std::random_device{}();
+  std::mt19937 gen(seed);
+  INFO("seed = " << seed);
+  std::uniform_real_distribution<> radius_dis(0.0, radius_of_sphere);
```

Comment:
Probably should stay away from the origin in the include_origin=false case.

Reply:
What value should the interior bound be in that case? `numeric_limits::epsilon * radius_of_sphere`?

Reply:
`test_jacobian` uses a finite difference stencil of half-width 3e-4.  If the map can't be evaluated at the origin it probably isn't numerically well-behaved near there, so we probably want to keep this stencil well away from r=0.  I wouldn't go less than 1e-3, probably even larger.

In any case, it might be a good idea to test a random point on this inner boundary, (similar to what is done for the outer boundary) not because the surface is intrinsically interesting, but because it would flush out RNG-induced test failures quicker.